numpy/lib/npyio.py

```diff
@@ -253,6 +255,138 @@ def iterkeys(self):
     def __contains__(self, key):
         return self.files.__contains__(key)
 
+class IncrementalWriter:
+    """
+    Class for the incremental writing of binary ``.npy`` files.
+    Multiple calls to ``save()`` append data to one binary ``.npy``
+    file resulting in one concatenated array saved. Saved arrays must
+    have exactly the same dtype and have compatible shapes. Shapes can
+    differ in the 1st dimension only which will be added up in the file.
+    The file is written in C order.
+
+    Parameters
+    ----------
+    fp : file-like object or string
+        Name of the file to open or a file-like object supporting
+        ``tell()``, ``seek()`` and ``write()``. In the latter case,
+        data will be written from current position.
+    flush : bool, optional
+        Calls ``flush()`` at the end of each ``save()``. Default:
+        False
+    hdrupdate : bool, optional
+        If True, updates header after each call to ``save()`` by seeking back
+        to the beginning (starting position) of the file, rewrites
+        header then seeks to the current position again.  Otherwise header
+        will be updated once at ``close()``.  Default: True.
```

Comment:
Would appreciate some rationale for why this is needed as well -- note that writing to the same piece of a file repeatedly is generally very cheap, because it just gets saved into the kernel's page buffers (RAM).

If it is needed, please call it `header_update` or similar. Or maybe `keep_header_synced`? Something using real words, anyway :-)